### ISSUE TITLE
Add method to get full match from `Captures`

### ISSUE THREAD

**Issue by tmccombs** [rust-lang/regex#1146]
#### Describe your feature request

Add a method to `regex::Captures` that is equvalent to (or even just this implementation):

```rust

#[inline]
fn full_match(&self) -> Match<'h'> {
    self.get(0).unwrap()
}
```

Getting the full match is a very common operation, and although the documetation for `Regex::get` says that getting the result at index 0 will always return `Some`, that property isn't reflected in the type system. And adding a method like this would expose that guarantee through the type of the public API.

**Comment by BurntSushi** [comment]
I think this is probably a good idea. I added an analogous method (among others) to the lower level [`regex-automata::util::captures::Captures`](https://docs.rs/regex-automata/latest/regex_automata/util/captures/struct.Captures.html#method.get_match) type.

I'm not sure `full_match` is the right name though. Hmmm. I'm not a huge fan of `get_match` either in this context. Probably `match` would be ideally, except for the obvious drawback that `match` is a Rust keyword.

**Comment by tmccombs** [comment]
Yeah, I'm not sure `full_match` is a great name either, but I couldn't think of a better one.

**Comment by ultrabear** [comment]
A potential other name could be `as_match`
